onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -1114,6 +1193,38 @@ def _infer_ZipMap(self, node):
         vi = self.known_vi_[node.output[0]]
         vi.CopyFrom(new_vi)
 
+    def _infer_Attention(self, node):
+        shape = self._get_shape(node, 0)
+        shape_bias = self._get_shape(node, 2)
+        shape[2] = shape_bias[0] / 3
+        output_dtype = self.known_vi_[node.input[0]].type.tensor_type.elem_type
+        vi = self.known_vi_[node.output[0]]
+        vi.CopyFrom(helper.make_tensor_value_info(node.output[0], output_dtype, shape))
+
+    def _infer_BiasGelu(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_FastGelu(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_Gelu(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_LayerNormalization(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_LongformerAttention(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_SkipLayerNormalization(self, node):
+        self._infer_BertOp_SameInputOutputShape_Impl(node)
+
+    def _infer_BertOp_SameInputOutputShape_Impl(self, node):
```

Comment:
This is general shape inference API and not limited to Bert ops. How about name like: 
```
def propagate_shape_and_type(self, node, input_index=0, output_index=0):
```